File: ilias-connector/ilias-connector-api/src/main/java/de/adesso/iliasdownloader3/service/exception/IliasException.java
package de.adesso.iliasdownloader3.service.exception;

public class IliasException extends RuntimeException {

    private static final long serialVersionUID = 168546164616546L;

    public IliasException(String message) {
        super(message);
    }

    public IliasException(String message, Throwable t) {
        super(message, t);
    }

    public IliasException(Throwable t) {
        super(t);
    }
}

File: src/main/kotlin/com/github/thetric/iliasdownloader/connector/api/exception/IliasException.kt
package com.github.thetric.iliasdownloader.connector.api.exception

/**
 * Base class for all Ilias exceptions.
 */
open class IliasException : RuntimeException {
    constructor(message: String) : super(message)

    constructor(message: String, t: Throwable) : super(message, t)

    constructor(t: Throwable) : super(t)
}
